### PR TITLE
Update apigatewayv2 request context

### PR DESCRIPTION
**Pull request by Tolledo**
*Description of changes:*
On Sept 9th,[ API Gateway launched support](https://aws.amazon.com/about-aws/whats-new/2020/09/api-gateway-http-apis-now-supports-lambda-and-iam-authorization-options/) for customers to secure Amazon API Gateway HTTP APIs using two new authorization options: Lambda authorizers and IAM authorizers. These new options enable customers to make flexible authorization decisions by providing an AWS Lambda function, or leveraging AWS IAM policies to control access to their APIs without writing any code. 

Updating API Gateway v2 event to support new context variables.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: events/apigw.go [hunk 1]
```diff
@@ -89,6 +91,24 @@ type APIGatewayV2HTTPRequestContextAuthorizerJWTDescription struct {
 	Scopes []string          `json:"scopes,omitempty"`
 }
 
+// APIGatewayV2HTTPRequestContextAuthorizerIAMDescription contains IAM information for the request context.
+type APIGatewayV2HTTPRequestContextAuthorizerIAMDescription struct {
+	AccessKey       string                                                  `json:"accessKey"`
+	AccountId       string                                                  `json:"accountId"`
```
Comment (bmoffatt) [inline events/apigw.go:97]: Naming should be `AccountID`

Comment (bmoffatt) [reply events/apigw.go:97]: ditto for all other "ID" fields added

File: events/apigw.go [hunk 2]
```diff
@@ -89,6 +91,24 @@ type APIGatewayV2HTTPRequestContextAuthorizerJWTDescription struct {
...
+	CallerId        string                                                  `json:"callerId"`
+	CognitoIdentity APIGatewayV2HTTPRequestContextAuthorizerCognitoIdentity `json:"cognitoIdentity,omitempty"`
+	PrincipalOrgId  string                                                  `json:"principalOrgId"`
+	UserArn         string                                                  `json:"userArn"`
```
Comment (bmoffatt) [inline events/apigw.go:101]: naming should be `UserARN`

File: events/apigw.go [hunk 3]
```diff
@@ -89,6 +91,24 @@ type APIGatewayV2HTTPRequestContextAuthorizerJWTDescription struct {
...
+	UserId          string                                                  `json:"userId"`
+}
+
+// APIGatewayV2HTTPRequestContextAuthorizerCognitoIdentity contains Cognito identity information for the request context.
+type APIGatewayV2HTTPRequestContextAuthorizerCognitoIdentity struct {
+	Amr            []string `json:"amr"`
```
Comment (bmoffatt) [inline events/apigw.go:107]: Q: What does AMR stand for? 

Capitalization should probably be `AMR`

File: events/apigw.go [hunk 4]
```diff
@@ -80,7 +80,9 @@ type APIGatewayV2HTTPRequestContext struct {
 
 // APIGatewayV2HTTPRequestContextAuthorizerDescription contains authorizer information for the request context.
 type APIGatewayV2HTTPRequestContextAuthorizerDescription struct {
-	JWT APIGatewayV2HTTPRequestContextAuthorizerJWTDescription `json:"jwt"`
+	JWT    *APIGatewayV2HTTPRequestContextAuthorizerJWTDescription `json:"jwt,omitempty"`
+	Lambda map[string]interface{}                                  `json:"lambda,omitempty"`
```
Comment (bmoffatt) [inline events/apigw.go:84]: Lambda authorizers can return plain strings... or ints, bools, arrays - right?

Would `interface{}` be more correct here?

**Review by bmoffatt — CHANGES_REQUESTED** [review]
A few naming conventions need to be fixed. One open question about the Lambda authorizer

File: events/apigw.go (hunk 4 above)
Comment (Tolledo) [reply events/apigw.go:84]: Authorizer always returns k/v pair, where key will always be a `string`.

**Review by bmoffatt — APPROVED** [review]

**Review by bmoffatt — APPROVED** [review]

**Pull request merged**